app/Http/Controllers/Api/CourseController.php

```diff
@@ -336,6 +335,46 @@ public function getStudentEnrolledCourses()
         return response()->json(['enrolledCourses' => $data], 200);
     }
 
+    public function courseStudents(){
+        $courseStudents = Course::with('students')->get();
+        return response()->json($courseStudents, 200);
+    }
+
+    public function courseStudentById(Request $request)
+    {
+        $validator = Validator::make($request->all(), [
+            'course_id' => 'required'
+        ]);
+
+        if ($validator->fails()) {
+            return response()->json(['error' => "course_id field is required"], 400);
+        }
+
+        $courseId = $request->get('course_id');
+        $courseStudent = Course::where('id',$courseId)->with('students')->withCount('students')->get();
+        if($courseStudent){
+            return response()->json($courseStudent, 200);
+        }
+        return response()->json(['success' => false], 400);
```

Comment:
Why is success having a false returned and 400 status 😩😩😩😩.  Is PHP confusing me ?

Reply:
Lol. Okay, so success is just a key right, and it's value is a Boolean, say can be successful or not.
Though, if you think there is a better approach to this, of cause we can implement that.

Reply:
Oh cool, the explanation makes sense. we go this route.